assignments/pull_requests.go

```
@@ -0,0 +1,157 @@
+		Organization: course.GetOrganizationPath(),
+		Repository:   repo.Name(),
+		Number:       int(pullRequest.GetNumber()),
+		Reviewers:    reviewers,
+	}
+
+	if err := sc.RequestReviewers(ctx, opt); err != nil {
+		return err
+	}
+	// Change pull request stage to review
+	pullRequest.SetReview()
+	return db.UpdatePullRequest(pullRequest)
+}
+
+// getNextReviewer gets the next reviewer from either teacherReviewCounter or studentReviewCounter,
+// based on whoever in total has been assigned to the least amount of pull requests.
+// It is simple, and does not account for how many current review requests any user has.
+//
+// Returns an error if the list of users is empty.
+func getNextReviewer(ID uint64, users []*pb.User, reviewCounter map[uint64]map[uint64]int) (*pb.User, error) {
+	if len(users) == 0 {
+		return nil, errors.New("list of users is empty")
+	}
+	reviewerMap, ok := reviewCounter[ID]
+	if !ok {
+		// If a map does not exist for a given ID we create it,
+		// and return the first user.
+		reviewCounter[ID] = make(map[uint64]int)
+		reviewCounter[ID][users[0].GetID()] = 1
+		return users[0], nil
+	}
+	userWithLowestCount := users[0]
+	lowestCount := reviewerMap[users[0].GetID()]
+	for _, user := range users {
+		count, ok := reviewerMap[user.GetID()]
+		if !ok {
+			// If the user is not present in the review map,
+			// then they are returned as the next reviewer.
+			reviewerMap[user.GetID()] = 1
+			return user, nil
+		}
+		if count < lowestCount {
+			userWithLowestCount = user
+			lowestCount = count
+		}
+	}
+	reviewerMap[userWithLowestCount.GetID()]++
+	return userWithLowestCount, nil
+}
+
+// getNextTeacherReviewer gets the teacher with the least total reviews.
+func getNextTeacherReviewer(db database.Database, course *pb.Course) (*pb.User, error) {
+	teachers, err := db.GetCourseTeachers(course)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get teachers from database: %w", err)
+	}
+	if len(teachers) == 0 {
```

Same reasoning as below. A course must always have a teacher, and `db.GetCourseTeachers()` should return an error if it doesn't.